tests/Integration/UI/SettingsPageTest.php

```diff
@@ -44,153 +44,185 @@ public function set_up(): void {
+
+		// Now try to set it back to empty values.
 		$options = self::$parsely->get_options();
 
-		$options['apikey'] = 'mydomain.com';
-		$expected          = $options;
+		$options['apikey']     = '';
+		$options['api_secret'] = '';
 
 		$actual = self::$settings_page->validate_options( $options );
-		self::assertSame( $expected, $actual );
+		self::assertSame( $options, $actual );
 	}
 
 	/**
-	 * Verifies that invalid Site ID values are emptied when validated.
+	 * Verifies that valid API credentials are retained when validated with the Validation API
 	 *
-	 * @since 3.9.0
+	 * @since 3.11.0
 	 *
 	 * @covers \Parsely\UI\Settings_Page::validate_basic_section
-	 * @uses \Parsely\Parsely::__construct
-	 * @uses \Parsely\Parsely::api_secret_is_set
-	 * @uses \Parsely\Parsely::are_credentials_managed
-	 * @uses \Parsely\Parsely::get_api_secret
-	 * @uses \Parsely\Parsely::get_options
-	 * @uses \Parsely\Parsely::get_managed_credentials
-	 * @uses \Parsely\UI\Settings_Page::__construct
-	 * @uses \Parsely\UI\Settings_Page::get_logo_default
-	 * @uses \Parsely\UI\Settings_Page::get_obfuscated_value
-	 * @uses \Parsely\UI\Settings_Page::get_unobfuscated_value
-	 * @uses \Parsely\UI\Settings_Page::sanitize_site_id
-	 * @uses \Parsely\UI\Settings_Page::validate_advanced_section
-	 * @uses \Parsely\UI\Settings_Page::validate_options
-	 * @uses \Parsely\UI\Settings_Page::validate_options_post_type_tracking
-	 * @uses \Parsely\UI\Settings_Page::validate_recrawl_section
-	 * @uses \Parsely\Validator::validate_site_id
+	 * @uses   \Parsely\Parsely::__construct
+	 * @uses   \Parsely\Parsely::api_secret_is_set
+	 * @uses   \Parsely\Parsely::are_credentials_managed
+	 * @uses   \Parsely\Parsely::get_api_secret
+	 * @uses   \Parsely\Parsely::get_options
+	 * @uses   \Parsely\Parsely::get_managed_credentials
+	 * @uses   \Parsely\UI\Settings_Page::__construct
+	 * @uses   \Parsely\UI\Settings_Page::get_logo_default
+	 * @uses   \Parsely\UI\Settings_Page::get_obfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::get_unobfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::sanitize_site_id
+	 * @uses   \Parsely\UI\Settings_Page::validate_advanced_section
+	 * @uses   \Parsely\UI\Settings_Page::validate_options
+	 * @uses   \Parsely\UI\Settings_Page::validate_options_post_type_tracking
+	 * @uses   \Parsely\UI\Settings_Page::validate_recrawl_section
+	 * @uses   \Parsely\Validator::validate_api_credentials
 	 *
 	 * @group settings-page
 	 * @group settings-page-validation
 	 */
-	public function test_invalid_site_id_values_are_emptied_when_validated(): void {
-		$expected = self::$parsely->get_options();
-		$options  = self::$parsely->get_options();
+	public function test_valid_api_credentials_are_retained_when_validated(): void {
+		$options = self::$parsely->get_options();
 
-		$options['apikey'] = 'invalid';
+		$options['apikey']     = 'mydomain.com';
+		$options['api_secret'] = 'valid_api_secret_key_based_on_length';
```

Comment:
```suggestion
		$options['api_secret'] = 'valid_api_secret';
```
Let's remove the "based on length" part since validation isn't being done by length anymore.